### ISSUE TITLE
E3514 support variable prefix

### ISSUE THREAD

**Issue by iann0036** [aws-cloudformation/cfn-lint#4545]
### CloudFormation Lint Version

cfn-lint 1.46.0

### What operating system are you using?

Mac

### Describe the bug

<img width="762" height="165" alt="Image" src="https://github.com/user-attachments/assets/89adf9e7-6d65-413b-81dc-601dfd24241f" />

### Expected behavior

When a variable prefixes the Sub, skip this rule.

### Reproduction template

```yaml
  WebAssetsBucket:
    Type: AWS::S3::Bucket
  BucketPolicy:
    Type: AWS::S3::BucketPolicy
    Properties:
      Bucket: abc123
      PolicyDocument:
        Statement:
          - Sid: BlockAllnonHttps
            Action: s3:*
            Condition:
              Bool:
                aws:SecureTransport: 'false'
            Effect: Deny
            Principal: '*'
            Resource:
              Fn::Sub:
                - ${S3Bucket_Arn}/*
                - S3Bucket_Arn:
                    Fn::GetAtt: WebAssetsBucket.Arn
```

**Comment by kddejong** [comment]
Thanks for filing this. I've tested your reproduction template against both v1.46.0 and the latest HEAD and wasn't able to reproduce the E3514 error — `Fn::GetAtt` values in Sub variable mappings are unresolvable, so the pattern check never fires.

Could you confirm this still reproduces for you on the latest version? If so, could you share the exact `cfn-lint` command and full error output? It's possible the actual template triggering this differs slightly from the reproduction case.

**Comment by iann0036** [comment]
Hey @kddejong,

Apologies, this seems to only occur with LanguageExtensions transform set. Prepend:

```
AWSTemplateFormatVersion: "2010-09-09"
Description: S3 buckets
Transform: AWS::LanguageExtensions

Resources:
```
